src/components/ContractPanel.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from "react";
+import { Mesh } from "../contracts/Mesh";
+import { Button, MintModal, BurnModal } from "../components";
+import { shortenAddress } from "../utils/shortenAddress";
+
+type Props = {
+  contract: Mesh;
+  address: string;
+};
+
+const ContractPanel: React.FC<Props> = (props) => {
+  const [authed, setAuthed] = useState(false);
+  const [error, setError] = useState<string>();
+  const [loading, setLoading] = useState(false);
+
+  const [isMintModalOpen, setIsMintModalOpen] = useState(false);
+  const [isBurnModalOpen, setIsBurnModalOpen] = useState(false);
+
+  const { contract, address } = props;
+
+  const authenticate = () => {
+    try {
+      const balance = contract.balanceOf(address);
+
+      setLoading(true);
+
+      balance.then((result) => {
+        if (+result < 1) return setError("No meshies owned by your address!");
+        setError(undefined);
+        return setAuthed(true);
+      });
+    } catch (err: any) {
+      setError(err as string);
+      return setAuthed(false);
```

Comment:
this error almost certainly is not a string, so it is not a safe typecast

Reply:
also, I think returning setAuthed will not allow the finally statement to run

Reply:
1. good catch - accidentally left over from previous iteration.
2. return value is saved and finally block is run, and then the return statement is